src/fmeval/eval_algorithms/faithfulness.py

```diff
@@ -0,0 +1,297 @@
+import logging
+from typing import Optional, List, Dict, Any
+
+from fmeval.constants import (
+    DatasetColumns,
+    MEAN,
+)
+from fmeval.data_loaders.util import get_dataset
+from fmeval.data_loaders.data_config import DataConfig
+from fmeval.eval_algorithms.save_strategy import SaveStrategy
+from fmeval.eval_algorithms.util import get_dataset_configs
+from fmeval.eval_algorithms.common import evaluate_dataset, get_default_judge_model
+from fmeval.eval_algorithms.eval_algorithm import EvalAlgorithmInterface, EvalAlgorithmConfig
+from fmeval.eval_algorithms import (
+    EvalAlgorithm,
+    EvalOutput,
+    EvalScore,
+)
+from fmeval.eval_algorithms.util import validate_dataset
+from fmeval.model_runners.model_runner import ModelRunner
+from fmeval.transforms.common import GetModelOutputs, GeneratePrompt
+from fmeval.transforms.transform import Transform
+from fmeval.transforms.transform_pipeline import TransformPipeline
+from fmeval.transforms.util import validate_call
+from fmeval.util import get_eval_results_path
+
+logger = logging.getLogger(__name__)
+
+FAITHFULNESS = EvalAlgorithm.FAITHFULNESS.value
+
+
+LONG_FORM_ANSWER_PROMPT = """\
+Human: Given a question and answer, create one or more statements from each sentence in the given answer. For each statement, please start with "Statement:".
+question:$question
+answer: $answer
+statements:\n Assistant:"""  # noqa: E501
+
+
+NLI_STATEMENTS_MESSAGE = """
```

Comment:
Isn't this instruction too complicated? I think it might be prudent if we can simplify the instruction. For example, we could provide a context, a question and ask the judge to answer one question, instead of asking 5 questions at once, and answer all 5 at once in specified manner. This might help reduce the chances of hallucination, and incorrect answers. 

These are just my thoughts. I maybe mistaken about this. There might be a reason why it is done this way. We can discuss in detail.

Reply:
Those questions can be deleted, it's examples included in the prompt to be feed to the judge model. If you see the placeholder, only one model_input will be replaced each time.

Reply:
I see. Where did you get this from? 
TBH, even I found it difficult to comprehend. I wonder if it is correct to expect all judge models to interpret correctly.

Reply:
The prompt in [LONG_FORM_ANSWER_PROMPT](https://github.com/xiaoyi-cheng/fmeval/blob/e49a89ab169055a68743d6af9def36c708854f4a/src/fmeval/eval_algorithms/faithfulness.py#L32) mentions: 
> For each statement, please start with "Statement:"
However, the few shot example here does not have that prefix for each statement. Could be nice to add that in for consistency.

Reply:
The prompt here has only 1 few-shot example to show the output structure of verdicts from the judge model. Adding in more examples could ensure that the judge model correctly follows the instruction. Ultimately, this is dependent on the choice of the judge model and will need some prompt engineering. The default prompt can be optimized by doing some prompt engineering experiments on the default judge model.